src/components/atoms/InputField/InputField.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { forwardRef } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+
+import { isTruthy } from "utils/types";
+
+import "./InputField.scss";
+
+export interface InputFieldProps extends React.HTMLProps<HTMLInputElement> {
+  iconStart?: IconProp | JSX.Element;
+  iconEnd?: IconProp | JSX.Element;
+}
+
+const isJsxElement = (
+  component: IconProp | JSX.Element
+): component is JSX.Element => {
+  return isTruthy(component?.hasOwnProperty("props"));
+};
+
+const renderIcon = (
+  icon: IconProp | JSX.Element,
+  modifiedClassName: string
+): JSX.Element => {
+  const iconComponent = isJsxElement(icon) ? (
+    icon
+  ) : (
+    <FontAwesomeIcon icon={icon} />
+  );
+
+  return (
+    <div className={classNames("input-field__icon", modifiedClassName)}>
+      {iconComponent}
+    </div>
+  );
+};
+
+const InputField: React.ForwardRefRenderFunction<
+  HTMLDivElement,
+  InputFieldProps
+> = ({ className, iconStart, iconEnd, ...extraInputProps }, ref) => {
```

Comment:
As per the other comment in this review, I would probably just have added `forwardRef` directly on this line as well, rather than including it as a `default` export below.